src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs

```diff
@@ -126,60 +123,75 @@ internal override DiagnosticInfo ErrorInfo
         /// <summary>
         /// Represents not nested missing type.
         /// </summary>
-        internal class TopLevel : MissingMetadataTypeSymbol
+        internal sealed class TopLevel : MissingMetadataTypeSymbol
         {
             private readonly string _namespaceName;
             private readonly ModuleSymbol _containingModule;
+            private readonly bool _isNativeInt;
```

Comment:
nit: How did you decide on whether error native ints should be error types with a flag versus wrapped error types?
I somewhat expected you'd be wrapping error types that are native ints, the same way we used to wrap tuple error types (if I recall correctly)?

Reply:
It was somewhat arbitrary to use a flag here (with the same underlying type) rather than a distinct type, although I believe there was a code path that became simpler as a result. We should go with a single approach for all cases. As we discussed offline, I will investigate using a flag in all cases in a subsequent PR.

---
In reply to: [374997101](https://github.com/dotnet/roslyn/pull/40994#discussion_r374997101) [](ancestors = 374997101)